tests/browser/page/browse.test.ts

```diff
@@ -42,7 +42,7 @@ describe("Page", () => {
       }
     );
 
-    // @ts-ignore
+    // @ts-expect-error - Throwing some weird type inference error due to ObjectiveFailed
```

Comment:
fucking unions

Reply:
I need to figure out a good way around this

Reply:
where the shape changes depending on what value it is? you do it by specifying it as T and conditionally defining it. See https://github.com/hdresearch/ts-sdk/blob/ce2daccd9c490acf8fbb54ef0b961efc31ce5335/src/types/index.ts#L36 and https://github.com/hdresearch/ts-sdk/blob/ce2daccd9c490acf8fbb54ef0b961efc31ce5335/src/types/index.ts#L5